core/Config.php

```diff
@@ -225,8 +225,17 @@ public static function getByDomainConfigPath()
      */
     public static function getHostname()
     {
+        $checkIfTrusted = true;
+        
+        try {
+            // this fails before the config was initialized
+            Config::getInstance()->existsLocalConfig();
```

Comment:
Not sure how often we are calling `getHostname` in the code, but this would actually add an additional file operation each time it's called. Wondering if we should check that another way if possible or maybe cache the result, so we don't check if the file exists each time it is called?

Reply:
I will rethink this and try to find a proper way to check if the config loaded instead of a `try...catch`

Reply:
@flamisz it's not about whether config file exists here it's more about checking if config has been loaded yet. Using Config::getInstance() here could actually result in a never ending loop as the getHostname method could be called while loading the config (might not be the case but it could be). 

I'm not sure it's possible to know if config has been loaded yet or not and in worst case we have to add a parameter `checkIfTrusted`